src/cryptoadvance/specter/controller.py

```diff
@@ -1225,10 +1225,20 @@ def wallet_receive(wallet_alias):
     # check that current address is unused
     # and generate new one if it is
     wallet.check_unused()
+    history_idx = int(request.args.get("history_idx", default=0))
+    past_addresses = wallet.addresses[0:-1][::-1][
+        (10 * history_idx) : (10 * (history_idx + 1))
+    ]
```

Comment:
No need to slice the array 3 times
```suggestion
    past_addresses = wallet.addresses[
        - 10 * history_idx - 2 : - 10 * (history_idx + 1) - 2: -1
    ]
```